### PR TITLE
Add SC calls handling for ETH->MvX 

### PR DESCRIPTION
**Pull request by ccorcoveanu**
The flow should be as follows: 

Since ETH contracts are non-upgradable we introduced a new proxy contract in front of the SafeContract.
This contract emits an event with some metadata we need for smart contract execution on MvX (tx data and gas limit + matching info like deposit and batch nonces) then forwards the deposit call to SafeContract.

### CODE REVIEW

File: integrationTests/mock/ethereumChainMock.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package mock
 import (
 	"context"
 	"fmt"
+	"github.com/ethereum/go-ethereum"
```
Comment (iulianpascalau) [inline integrationTests/mock/ethereumChainMock.go:6]: go imports

File: integrationTests/mock/ethereumChainMock.go [hunk 2]
```diff
@@ -252,6 +254,14 @@ func (mock *EthereumChainMock) BalanceAt(ctx context.Context, account common.Add
 	return big.NewInt(0), nil
 }
 
+func (mock *EthereumChainMock) FilterLogs(ctx context.Context, q ethereum.FilterQuery) ([]types.Log, error) {
```
Comment (iulianpascalau) [inline integrationTests/mock/ethereumChainMock.go:257]: missing comments

File: testsCommon/bridge/ethereumClientStub.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package bridge
 
 import (
 	"context"
+	"github.com/multiversx/mx-bridge-eth-go/clients/ethereum/contract"
```
Comment (iulianpascalau) [inline testsCommon/bridge/ethereumClientStub.go:5]: go imports

File: testsCommon/bridge/ethereumClientStub.go [hunk 4]
```diff
@@ -100,6 +103,22 @@ func (stub *EthereumClientStub) IsQuorumReached(ctx context.Context, msgHash com
 	return false, errNotImplemented
 }
 
+func (stub *EthereumClientStub) IsDepositSCCall(deposit *clients.DepositTransfer) bool {
```
Comment (iulianpascalau) [inline testsCommon/bridge/ethereumClientStub.go:106]: missing comments here and on L114

File: testsCommon/bridge/ethereumClientWrapperStub.go [hunk 5]
```diff
@@ -3,6 +3,7 @@ package bridge
 import (
 	"context"
 	"errors"
+	"github.com/ethereum/go-ethereum"
```
Comment (iulianpascalau) [inline testsCommon/bridge/ethereumClientWrapperStub.go:6]: go imports

File: testsCommon/bridge/scExecProxyContractStub.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// SCExecProxyContractStub -
+type SCExecProxyContractStub struct {
+	FilterLogsCalled func(ctx context.Context, q ethereum.FilterQuery) ([]types.Log, error)
+}
+
+func (stub *SCExecProxyContractStub) FilterLogs(ctx context.Context, q ethereum.FilterQuery) ([]types.Log, error) {
```
Comment (iulianpascalau) [inline testsCommon/bridge/scExecProxyContractStub.go:14]: missing comments

File: testsCommon/interactors/blockchainClientStub.go [hunk 7]
```diff
@@ -2,6 +2,8 @@ package interactors
 
 import (
 	"context"
+	"github.com/ethereum/go-ethereum"
```
Comment (iulianpascalau) [inline testsCommon/interactors/blockchainClientStub.go:5]: go imports

**Review by sasurobert — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**